packages/cli/src/runners/task-runner-process.ts

```diff
@@ -74,6 +86,20 @@ export class TaskRunnerProcess extends TypedEmitter<TaskRunnerProcessEventMap> {
 		);
 
 		this.logger = logger.scoped('task-runner');
+
+		this.runnerLifecycleEvents.on('runner:failed-heartbeat-check', () => {
+			this.logger.warn('Task runner failed heartbeat check, restarting...');
+			this.shouldForceRestartRunner = true;
```

Comment:
Would it be possible to do this without this flag? Could we perhaps add `restart()` method (which doesn't set the `isShuttingDown`) and use that instead of `stop()`?